roles/postgres/tasks/main.yml

```diff
@@ -42,28 +42,17 @@
     enabled: true
   become: true
 
-- name: Setup postgresql databases
-  community.postgresql.postgresql_db:
-    name: "{{ item.database }}"
-  with_items: "{{ postgres_users }}"
-  become: true
-  become_user: postgres
-
 - name: Setup postgresql users
   community.postgresql.postgresql_user:
-    db: "{{ item.database }}"
```

Comment:
This was useless, as the option controls which database to connect not, not which database grant permissions on. We connect as admin role `postgres` to database `postgres`, and that is sufficient to create user roles and databases.